### PR TITLE
Rest spec and documentation

### PR DESCRIPTION
**Pull request by jkakavas**
This change adds the spec for the new REST APIs that we
introduce for the IDP and documentation for each of the APIs. The
documentation pages are intentionally not included in the API
reference so as to minimize unnecessary exposure.

### CODE REVIEW

**Comment by tvernum** [comment]
I think we might need to consider whether these actually should be in the main docs folder or not. Some language clients are generated automatically from the specs, and we don't want them to add bindings for these APIs.

File: x-pack/plugin/identity-provider/docs/en/rest-api/idp-saml-register-sp.asciidoc [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+The following example registers a service provider for use with the identity provider
+
+[source, console]
+--------------------------------------------------------------------
+POST /_idp/saml/sp/https%3A%2F%2Fsome.sp.org%2F
+{
+  "name" : "My Service Provider",
+  "entity_id" : "https://some.sp.org/",
+  "acs" : "https://some.sp.org/saml/acs",
+  "privilegs" : {
```
Comment (lloydmeta) [inline x-pack/plugin/identity-provider/docs/en/rest-api/idp-saml-register-sp.asciidoc:90]:
```suggestion
  "privileges" : {
```

File: x-pack/plugin/identity-provider/docs/en/rest-api/idp-saml-init.asciidoc [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+IDP initiated or SP initiated SAML Single Sign On. This API expects the caller to present
+credentials for the user that the SAML Response will be created for as "Secondary Authentication"
+using the `es-secondary-authorization` HTTP Request header.
+
+The SAML response is returned as an XML String and the caller of the API is responsible to instruct
+the end user's browser to make an HTTP Post request to the Service Provider with the SAML response
+Base64 and URL encoded.
+
+[[idp-saml-init-body]]
+==== {api-request-body-title}
```
Comment (lloydmeta) [inline x-pack/plugin/identity-provider/docs/en/rest-api/idp-saml-init.asciidoc:32]: Is there a `acs` body param?

https://github.com/elastic/elasticsearch/pull/54215/files#diff-1cd906e941a844a669917a5383d63b97R127

Comment (lloydmeta) [reply x-pack/plugin/identity-provider/docs/en/rest-api/idp-saml-init.asciidoc:32]: Hmm, in `7.7-SNAPSHOT` if I pass the `acs` body param I get

```json
{
	"error": {
		"root_cause": [{
			"type": "x_content_parse_exception",
			"reason": "[1:34] [idp_init_sso] unknown field [acs]"
		}],
		"type": "x_content_parse_exception",
		"reason": "[1:34] [idp_init_sso] unknown field [acs]"
	},
	"status": 400
}
```

Comment (jkakavas) [reply x-pack/plugin/identity-provider/docs/en/rest-api/idp-saml-init.asciidoc:32]: The snapshot you are using does not contain https://github.com/elastic/elasticsearch/commit/5434a3c6714596b3947de3605d93285df5507ad6 ( was merged 6 hours ago) in which this change is added in 7.7 
These docs need updating after recent code changes Lloyd, I'll update the PR today and ping you when it's ready

Comment (lloydmeta) [reply x-pack/plugin/identity-provider/docs/en/rest-api/idp-saml-init.asciidoc:32]: Ah, thanks, just got a newer snapshot and it works now with the `acs` param.

**Pull request closed without merging**